### PR TITLE
Update squareio wire-schema

### PR DESCRIPTION
**Pull request by strokyl**
*Issue*

This library depends on wireschema 3.7.1 when confluent protobuf deserializer depends on 4.3.0. This make it impossible to use both in the same project.
Here is my proposal to update wireschema to make both libraries compatible.
I didn't update to 4.4.3 (latest) because there is also breaking change between 4.3.0 and 4.4.0

*Description of changes:*

- use okio FakeSystem instead of jimfs ProtobufSchemaLoader (see: https://github.com/square/wire/pull/1918)

### CODE REVIEW

**Comment by franoisD** [comment]
image

File: serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/utils/apicurio/FileDescriptorUtils.java
Comment (YangSan0622) [inline serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/utils/apicurio/FileDescriptorUtils.java:101]: Can you revert these formatting change since it increases the difficultly of code review?

Comment (strokyl) [reply serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/utils/apicurio/FileDescriptorUtils.java:101]: Do you have any formatting tools in place to help me?

Comment (strokyl) [reply serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/utils/apicurio/FileDescriptorUtils.java:101]: Otherwise adding `&w=1` in the review URL will only show diff without space. https://github.blog/2018-05-01-ignore-white-space-in-code-review/

Comment (strokyl) [reply serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/utils/apicurio/FileDescriptorUtils.java]: I have just remove the whitespace anyway

**Comment by strokyl** [comment]
up

**Comment by strokyl** [comment]
up

**Comment by wavejumper** [comment]
Just chiming in and saying that the use of 3.7.1 of wire-schema is also impacting our use of this library. We also have a dependency on Confluent's schema registry in our project. 
We depend on features in later versions of Confluent's schema registry, but can't bump because of this conflict.

Re CVE: while I understand from #235 that this library isn't impacted by this vulnerability, it would still be a plus to bump this dependency. For our use case, we have customers who consume our built artefact (a JAR file) and run their own CVE scanning on it. We try to strive to have as close to zero CVEs in our product.
Bumping wire-schema would also mean removing many suppressions from our nvd xml file (another win).

**Review by YangSan0622 — CHANGES_REQUESTED** [review]
Sorry for the late update, I have went through the code and LGTM.
Do you mind resolve the merge confilct and squash the commits? I can approve the PR after that.

**Comment by strokyl** [comment]
> Sorry for the late update, I have went through the code and LGTM. Do you mind resolve the merge confilct and squash the commits? I can approve the PR after that.

Done

**Review by YangSan0622 — APPROVED** [review]

**Pull request merged**
